include/ray.h

```diff
@@ -2,19 +2,21 @@
 #define RAY_H
 
 #include "commons.h"
+// TODO: Can't forward declare due to destructorm, make smart pointer
+#include "intersection_point.h"
 
 class Ray
 {
 private:
   Vertex v_start_;
   Vertex v_end_;
-
   ColorDbl color_;
-
-  void* intersecting_triangle_;
+  // TODO: make smart pointer
+  IntersectionPoint* intersection_point_ = nullptr;
 
 public:
   Ray() {}
+  ~Ray() { if(intersection_point_) delete intersection_point_; }
```

Comment:
Personal preference: ALWAYS encapsule the scope of if's, for's and while's etc. with {}